### ISSUE TITLE
[Feature] Region folding support

### ISSUE THREAD

**Issue by jefferyharrell** [jonnyboyC/kos-language-server#77]
If I'm not mistaken, VS Code now supports region folding, but it has to be implemented per language extension. It'd be great if the kOS language mode supported a syntax something like

```
//#region somename

// arbitrary amount of code

//#endregion
```

I have no idea how to develop for VS Code so I can't offer any implementation suggestions, but [here's a link that might be a place to start](https://code.visualstudio.com/updates/v1_17#_folding-regions).

Thanks!

**Comment by jonnyboyC** [comment]
Thanks for the feature request! I think this would definitely be doable, any preference in terms of what the folding region be called. I'm perfectly fine with `#region` / `#endregion` as tags or anything else. I may ask on the kOS discord if anyone else has a preference as well.

**Comment by jefferyharrell** [comment]
Thanks for the extension! I've really been relying heavily on it, as I've been working on a kOS project for fun in my spare time.

As for the syntax, I think "//#region" and "//#endregion" would be good choices because they're the same as VS Code's built-in JavaScript region folding syntax. But by all means, ask around. I don't have a strong opinion. I've just got a monster kOS file and I'd like to scroll less. ;-)

**Comment by jonnyboyC** [comment]
Just released version 0.8.0 With the new folding region. No one in particularly had a preference so this will work now
    
    // #region 
    print("foldable").
    // #endregion
